### ISSUE TITLE
App freezes completely when attempting to save a new file

### ISSUE THREAD

**Issue by block36underscore** [JannisX11/blockbench#2364]
### What are you trying to do, and what do you expect to happen?

I am trying to save a file. The file explorer menu should close once the location is selected and the app should continue working as normal. The file should also save.

### What happens instead?

The file is not saved, and the app does not respond to any inputs.

I have tried this on both the deb installation, the flatpak installation, and the appimage. All of them act the same. I was seeing some weird messages from AppArmor in dmesg, but they were inconsistent, and unrelated to file saving. No messages show up in the electron console, or the one running the app.

### Model format in which the issue occurs

I have tested generic and images, but it probably happens to all of them.

### Blockbench variant

Program

### Blockbench version

4.10.2

### Operating System

Linux (Debian based)

### Installed Blockbench plugins

none

**Comment by block36underscore** [comment]
Restarting my computer resolved this issue

**Comment by block36underscore** [comment]
I spoke too soon. I reinstalled the deb version and this issue is back again. I'll be investigating some more

**Comment by block36underscore** [comment]
I spoke too soon. I reinstalled the deb version and this issue is back again. I'll be investigating this a bit more

**Comment by block36underscore** [comment]
After some testing I have found that saving works as normal, but file creation (or overwriting a file) is linked with the freeze

It may or may not be relevant, but I'm getting the weird AppArmor message again.
```
[   35.913174] audit: type=1400 audit(1717091973.756:229): apparmor="AUDIT" operation="userns_create" class="namespace" info="Userns create - transitioning profile" profile="unconfined" pid=4214 comm="blockbench" requested="userns_create" target="unprivileged_userns"
[   35.913928] audit: type=1400 audit(1717091973.757:230): apparmor="DENIED" operation="capable" class="cap" profile="unprivileged_userns" pid=4216 comm="blockbench" capability=21  capname="sys_admin"
[   37.249371] i915 0000:00:02.0: Using 39-bit DMA addresses
[   82.090824] audit: type=1400 audit(1717092019.744:231): apparmor="AUDIT" operation="userns_create" class="namespace" info="Userns create - transitioning profile" profile="unconfined" pid=4500 comm="blockbench" requested="userns_create" target="unprivileged_userns"
[   82.091539] audit: type=1400 audit(1717092019.744:232): apparmor="DENIED" operation="capable" class="cap" profile="unprivileged_userns" pid=4502 comm="blockbench" capability=21  capname="sys_admin"
```

**Comment by boot2big** [comment]
Same issue here, unsure of any way to remedy this on my end either. Behaves similarly in that I'm unable to create files, whether new or as an overwrite of existing ones.

**Comment by YTFrost** [comment]
Same issue on EndeavourOS (Arch-based). Getting the same errors.

**Comment by AlexTMjugador** [comment]
Happens over here on Debian Sid too, downgrading to 4.10.1 fixed my problem.

**Comment by block36underscore** [comment]
Downgrading to 4.10.1 also worked for me.

**Comment by boot2big** [comment]
Seems that's what I'll have to do in the meantime as well, though I'd much rather be able to run the latest stable version of BB for modeling.

**Comment by OrdinaryCloverOC** [comment]
same with flatpak/rpm on fedora

**Comment by Vetpetmon** [comment]
App also freezes on my system. Pop!_OS distro, deb package. 4.10.2

Attempted with generic model. forcing it to close resets all user preferences and plugins

Thankfully, it keeps a backup of the project, but the app is unusable otherwise.

**Comment by AlexTMjugador** [comment]
I couldn't reproduce the issue when building Blockbench from source via `npm ci && npm run dist` with the latest `master` branch code. That's another possible workaround, for people comfortable with building Node projects.

**Comment by halotroop2288** [comment]
Built from source (22e6b9c08e6b891fe85c4f877c2ae5ebed12d2ae) just to confirm that this is still not fixed. 👎🏻

**Comment by AlexTMjugador** [comment]
> Built from source ([22e6b9c](https://github.com/JannisX11/blockbench/commit/22e6b9c08e6b891fe85c4f877c2ae5ebed12d2ae)) just to confirm that this is still not fixed. 👎🏻

The `next` branch the commit you mentioned belongs to doesn't contain commit [`bdf0043`](https://github.com/JannisX11/blockbench/commit/bdf00433647bc24a2d74c8aadbcb673001c57bee), which is only available on `master`.

**Comment by halotroop2288** [comment]
Yes, the master branch does contain [that commit](https://github.com/JannisX11/blockbench/commit/bdf00433647bc24a2d74c8aadbcb673001c57bee).
![image](https://github.com/JannisX11/blockbench/assets/30361266/0454a02f-0563-4ea9-bd57-1217a63fb81e)

That is exactly the branch I cloned to build from source. The `next` branch has nothing to do with this.

![image](https://github.com/JannisX11/blockbench/assets/30361266/0e51b4ab-cce8-4951-b8a1-190aef713195)

**Comment by AlexTMjugador** [comment]
Okay... It looks like the v4.10.3 release didn't fix this issue for you then? On my end, it was fixed after updating to v4.10.3.

**Comment by halotroop2288** [comment]
4.10.3 is the only version I ever installed on my Linux system. I don't have the choice to go back, because downgrading packages tends to break Arch.

**Comment by JannisX11** [comment]
You won't receive the fix if you are trying to build from source using the regular npm scripts. Linux uses an older electron version does doesn't have the issue yet and it's switched out in the CI script.
If you are still having the issue, try downloading Blockbench from the website since third party builds may not contain the fix in some cases.

**Comment by halotroop2288** [comment]
> try downloading Blockbench from the website
That checks out. Now it's fixed.
> third party builds may not contain the fix in some cases.
Then can you please provide an official AUR package? I do not like installing apps outside of the official package manager.

**Comment by Des-Nerger** [comment]
> You won't receive the fix if you are trying to build from source using the regular npm scripts.

Would it please be notified here in this thread when that problem is finally resolved, so that we know when we can switch from v4.10.1 back to master?

**Comment by JannisX11** [comment]
This issue was resolved in third party builds with 4.10.4 on July 6th.
